src/main/java/br/edu/ufcg/computacao/eureca/backend/api/http/exceptions/pre_enrollment/StudentNotFoundException.java

```diff
@@ -0,0 +1,12 @@
+package br.edu.ufcg.computacao.eureca.backend.api.http.exceptions.pre_enrollment;
+
+import br.edu.ufcg.computacao.eureca.common.exceptions.EurecaException;
+
+public class StudentNotFoundException extends EurecaException {
+
+    private static final String STUDENT_NOT_FOUND = "O estudante com matrícula %s não está cadastrado no sistema.";
+
+    public StudentNotFoundException(String registration) {
+        super(String.format(STUDENT_NOT_FOUND, registration));
```

Comment:
No need for change the original exception handling code. You could simply customize the string where the exception in created (ex. line 86 in https://github.com/computacao-ufcg/eureca-backend/pull/138/files#diff-6d1d5471eda288bdcbb445d424146276f4ba586330a055bff38900c5b2614f34)

Reply:
I think is better to have specific exceptions, each one with a status code. If all the raised exceptions are `InvalidParameterException`, we only get one status code, which can be incorrect for the raised exception. For example: a not found student raising a error with the correct message, but the status code 400 (bad request), where should be 404.

Reply:
I am not following. The HTTP error codes are mapped to the different types of the basic EurecaException in the common package (if a new code is needed -- I doubt it, then, it is that class that needs to be enhanced): 

UnauthorizedRequestException ->  HttpStatus.FORBIDDEN
UnauthenticatedUserException -> HttpStatus.UNAUTHORIZED
ConfigurationErrorException -> HttpStatus.CONFLICT
InvalidParameterException -> HttpStatus.BAD_REQUEST
InternalServerErrorException -> INTERNAL_SERVER_ERROR
Exception -> UNSUPPORTED_MEDIA_TYPE

Then, the particular exceptions that signal a particular error must extend the appropriate general Eureca exception (as mentioned in a previous comment). The code in https://github.com/computacao-ufcg/eureca-backend/pull/138/files#diff-70d24fe9ceeeadaa90b6019384bf8077fd237ce8cc22f3a5ebe321780a135267 should the be something like:

...
@ExceptionHandler(StudentNotFoundException.class)
    public ResponseEntity<ExceptionResponse> handleStudentNotFound(EurecaException exception, WebRequest req) {
        return this.handleInvalidParameterException(exception, req);
    }
....

Reply:
I understood, but i still disagree about the HttpStatus.BAD_REQUEST for a not found student (or any other entity), because is semantically incorrect. The 400 status code means a syntax error in the request. So, the best solution would be create a `EntityNotFoundException` on eureca-common, and make these new Exceptions extend the new class?

Reply:
You're right. We need to create a new exception in common (NotFoundException) which should be mapped to HttpStatus.NOT_FOUND. Then, (backend) StudentNotFoundException should extend (common) NotFoundException. Does it make sense to you?

Reply:
Yes, total